repos/system_upgrade/el7toel8/actors/removeoldpammodulesapply/actor.py

```diff
@@ -0,0 +1,34 @@
+from leapp.actors import Actor
+from leapp.libraries.actor.library import comment_modules, read_file
+from leapp.libraries.common.pam import PAM
+from leapp.models import RemovedPAMModules
+from leapp.tags import IPUWorkflowTag, PreparationPhaseTag
+
+
+class RemoveOldPAMModulesApply(Actor):
+    """
+    Remove old PAM modules that are no longer available in RHEL-8 from
+    PAM configuration to avoid system lock out.
+    """
+
+    name = 'removed_pam_modules_apply'
+    consumes = (RemovedPAMModules,)
+    produces = ()
+    tags = (IPUWorkflowTag, PreparationPhaseTag)
+
+    def process(self):
+        model = next(self.consume(RemovedPAMModules))
+        files = {}
+
+        # Read files
```

Comment:
Is there actually any particular need for 3 loops over the same content?

The whole `process` method looks to me it could be reduced to this:
```python
for model in self.consume(RemovedPAMModules):
    for path in PAM.files:
        content = read_file(path)
        with open(path, 'w') as f:
            f.write(comment_modules(model.modules, content))
    break
```